### DISCUSSION TITLE
RAM Limit for community cluster?

### DISCUSSION THREAD

**Discussion by probonopd** [cirruslabs/cirrus-ci-docs#800] [Q&A]
https://cirrus-ci.org/faq/#are-there-any-limits explains CPU limits but not RAM limits.

So, what is the maximum allowable RAM size for builds on community clusters?

**Comment by probonopd** [comment]
Trial and error seems to suggest 13 GB for FreeBSD?

**Comment by fkorotkov** [comment]
For `container`s it's flexible and can go up to `90G`. For the VMs like `freebsd_instance` it falls under the [GCP's specifications](https://cloud.google.com/compute/docs/instances/creating-instance-with-custom-machine-type) of:

> 0.5 GB to 8 GB per vCPU

**Reply by probonopd** [reply]
Wow, interesting. Thanks for the expanation @fkorotkov, I would never have imagined that the maximum RAM is tied to the number of vCPUs.

The reason I was asking is because I need to use RAM-backed tmpfs in my jobs because the regular disk seems really very slow in `freebsd_instance`. Or is there a trick to speed it up? (The graphic in Cirrus CI seems to suggest that at best one of the two vCPUs I am using is maxed out.)

**Reply by fkorotkov** [reply]
For regular `container` [I recently tuned our GKE](https://twitter.com/fedor/status/1354505744293502980) which gave some [mind blowing performance results](https://twitter.com/fedor/status/1365389827772940288).

For VM-based instances, Cirrus was using the default [balanced disk type](https://cloud.google.com/compute/docs/disks/performance#type_comparison) that has a close to SSD performance but not quite on small disks (peak I/O also depends on a disk size similar to memory and vCPU dependency). I just switched all Community VMs to explicitly use `pd-ssd` disks so it might help with your `freebsd_instance` tasks. Please let me know if switching to SSD improved build times for you.

**Reply by probonopd** [reply]
Thank you very much @fkorotkov, the new setting seems to have improved performance a _lot_ :+1:

**Reply by fkorotkov** [reply]
@probonopd glad to hear it! Can I ask just for the record what kind of improvement did you see?

**Reply by probonopd** [reply]
_At least_ a factor of three. From unusable to usable. (Before this, I was using tmpfs because the disk was so slow; now I no longer need the tmpfs and the builds are only slightly slower than with the tmpfs. Sorry I don't have any measurements from before I started using tmpfs but it was unusably slow.)

**Reply by fkorotkov** [reply]
> unusable to usable

Is quite a measurement! 😅 Thanks! Glad to hear it.

**Reply by probonopd** [reply]
By the way, a __huge__ "Thank You" to Cirrus CI and you personally. Our project would not be possible without you. I hope it is OK to mention you in our [Acknowledgements](https://hellosystem.github.io/docs/developer/acknowledgements).

**Reply by fkorotkov** [reply]
Ahhh! I'm flattered! Always happy to help!
